### PR TITLE
Logrotate template and state file to handle log file rotation

### PR DESCRIPTION
**Pull request by shaidar**
#### What's this PR do?
Logrotate template and state file to handle log file rotation defined in pillar

### CODE REVIEW

File: salt/utils/logrotate.sls [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+{% for name, settings in logrotate_files.items() %}
+
+generate_{{ name }}_logrotate:
+  file.managed:
+    - name: /etc/logrotate.d/{{ name }}
+    - source: salt://utils/logrotate/conf.jinja
+    - template: jinja
+    - mode: '0644'
+    - context:
+        logrotate_files: {{ logrotate_files }}
```
Comment (blarghmatey) [inline salt/utils/logrotate.sls]: I'm assuming this should be the `settings` variable from the `.items()` iteration

File: salt/utils/logrotate/conf.jinja [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{%- for name, settings in logrotate_files.items() -%}
```
Comment (blarghmatey) [inline salt/utils/logrotate/conf.jinja]: I'm assuming this is intended to be `settings` instead of `logrotate_files`

Comment (shaidar) [reply salt/utils/logrotate/conf.jinja]: No, believe it is correct. A sample pillar file looks like the following:
```logrotate:
  test0:
    conf_file: /etc/logrotate.d/test0
    name: /var/log/test0.log
    options:
      - rotate 2
      - size 5k
      - missingok
     - postrotate systemctl restart nginx
```

**Comment by blarghmatey** [comment]
:+1: as long as the intent is to have one block per file

**Comment by shaidar** [comment]
Yeah think it might be best to have each service have its separate config file. That way, we can use same configs across multiple services.

**Pull request merged**